clients/spark/core/src/main/hadoop3/scala/io/treeverse/clients/conditional/S3ClientBuilder.scala

```diff
@@ -36,15 +38,50 @@ object S3ClientBuilder extends io.treeverse.clients.S3ClientBuilder {
         )
       } else None
 
+    initializeClient(configuration, credentialsProvider, region, bucket)
```

Comment:
This is a lot of code duplication between the Hadoop 2 and Hadoop 3 variations.  It suggests that We (in the sense of the royal "We", I'm ashamed to say) put the conditional API in the wrong place.  Can you please open an issue to refactor the API along more useful lines?  For instance we could have a method `prepareMostOfTheAWSClientBuilderButLeaveMeTheBitWhereISetTheCredentials` (name open to discussion) that is shared between all both variations.